### ISSUE TITLE
No 'chr' in front of chromosome names in vcf

### ISSUE THREAD

**Issue by ljohansson** [edg1983/GREEN-VARAN#3]
Hi,

Thank you for providing such a wonderful tool, combining so many tools to help in prioritizing non-coding variants. 
I am trying to run the GREEN-VARAN on a GRCh37 dataset. However, our reference genome labels the chromosomes as 1, 2, 3, etc. 
 GREEN-VARAN seems to expect chr1, chr2, chr3, etc. Is there an option to change this? I found the utils.nim file listing STDCHROMS, but changing the list there didn't change anything. 

In addition, is there also a GRCh37 test vcf available?

Regards,
Lennart Johansson

**Comment by edg1983** [comment]
Hi Lennart,

Thanks for reporting this. I think the problem is simply due to the default list of chromosome names that, as you noted, includes "chr" prefix.
The annotation process upstream of GREEN-VARAN is agnostic of "chr" prefix, and GREEN-VARAN itself only requires that they match between your VCF file and the GREEN-DB file used in the annotations. The GRCh37 version of GREEN-DB provided in the repository should contain standard chromosomes without "chr" so it should work fine in your case (please double check this in the downloaded file `GRCh37_GREEN-DB.bed.gz`). The problem is that when you do no specify any chromosome, GREEN-VARAN use the list in `STDCHROMS` to cycle through contigs in the VCF using an index query and thus it doesn't find any variant in your case.

To override the default chromosome names, you can use `--chrom` option. You can pass either a list of comma-separated values of a file containing 1 chromosome name per line.

I will fix the GREEN-DB query so that it would be agnostic to "chr"  prefix in the future so that the default would work for both GRCh37 adn GRCh38.

**Comment by ljohansson** [comment]
Hi Edoardo,

Thanks for your answer. The GRCh37_GREEN-DB.bed.gz file indeed has 'chr' as prefix. I have done several tests.
1. remove the chr prefix from the GRCh37_GREEN-DB.bed.gz file
2. add --chrom 1 to the command
3. changing the STDCHROMS to "1", "2", "3", etc.
4. combi 1, 2 and 3
However, the error stays the same. And even though only asking for chromosome 1 output the error remains mentioning all chromosomes.

[2022-03-15T11:50:54] - INFO: Reading config from file: prioritize_smallvars.json
[2022-03-15T11:50:54] - DEBUG: af:@["gnomAD_AF", "gnomAD_AF_nfe"]
[2022-03-15T11:50:54] - DEBUG: maxaf:0.009999999776482582
[2022-03-15T11:50:54] - DEBUG: regions:@["TFBS", "DNase", "UCNE"]
[2022-03-15T11:50:54] - DEBUG: scores:{"FATHMM_MKLNC": 0.9079999923706055, "ncER": 98.59999847412109, "ReMM": 0.9629999995231628}
[2022-03-15T11:50:54] - DEBUG: constraint:0.699999988079071
[2022-03-15T11:50:54] - DEBUG: more_regions:@[]
[2022-03-15T11:50:54] - DEBUG: more_values:{:}
[2022-03-15T11:50:54] - INFO: N selected chromosomes: 25
[2022-03-15T11:50:54] - INFO: N selected genes: 0
[2022-03-15T11:50:54] - DEBUG: Selected chromosomes: @["chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY", "chrM"]
[2022-03-15T11:50:54] - DEBUG: Selected genes: {}
[2022-03-15T11:50:54] - INFO: Update existing gene annotations: true
[2022-03-15T11:50:54] - INFO: Filter mode active: false
[2022-03-15T11:50:54] - INFO: === Start processing VCF ===
[2022-03-15T11:50:54] - WARN: No ANN or BCSQ field detected in header so ANN will be created
[2022-03-15T11:50:54] - DEBUG: Processing chromosome chr1
[2022-03-15T11:50:54] - DEBUG: chromosome chr1: 0 processed; 0 annotated, 0 vars of interest, 0 written
[2022-03-15T11:50:54] - DEBUG: Processing chromosome chr2
...
[2022-03-15T11:50:54] - DEBUG: chromosome chrX: 0 processed; 0 annotated, 0 vars of interest, 0 written
[2022-03-15T11:50:54] - DEBUG: Processing chromosome chrY
[2022-03-15T11:50:54] - DEBUG: chromosome chrY: 0 processed; 0 annotated, 0 vars of interest, 0 written
[2022-03-15T11:50:54] - DEBUG: Processing chromosome chrM
[2022-03-15T11:50:54] - DEBUG: chromosome chrM: 0 processed; 0 annotated, 0 vars of interest, 0 written
[2022-03-15T11:50:54] - INFO: 0 variants annotated with greendb information
[2022-03-15T11:50:54] - INFO: 0 vars of interest based on the input gene list if any
[2022-03-15T11:50:54] - INFO: 0 variants written to output
[2022-03-15T11:50:54] - INFO: All done - Completed in 0.0 min 0.02 sec

**Comment by edg1983** [comment]
Strange.

As you noted the GRCh37 version of the DB contains the chr prefix, so I've generated a new file without "chr" and teh corresponding index file as follows:
```
zcat GRCh37_GREEN-DB.bed.gz | sed 's/^chr//g' | bgzip -c > GRCh37_GREEN-DB.nochr.bed.gz
tabix -p bed --csi GRCh37_GREEN-DB.nochr.bed.gz
```

Then using the following command correctly annotated the chromosome 22 from a VCF without "chr" prefix
```
./greenvaran smallvars \
-i test_input_22.vcf.gz \
-o test_22.vcf.gz \
--chrom 22 \
--log test_22.log \
--dbschema GREEN-VARAN/config/greendb_schema_v2.5.json \
--config GREEN-VARAN/config/prioritize_smallvars.json \
--db GRCh37_GREEN-DB.nochr.bed.gz
```

**Comment by ljohansson** [comment]
This seems to work. Thank you.

**Comment by ljohansson** [comment]
Dear Edoardo,

I was wondering about the output vcf. In trying to understand the annotations and procedures greenvaran uses, I was looking at the vcf file for the annotations of the regions and scores. Per variant the green_db levels are annotated, however I am also interested in knowing the annotations of the specific databases and tools. Is there a way using the "greenvaran smallvars" command to also write the 'all' or 'best' annotations to the vcf file. Or should I switch to the 'main.nf' way of starting the tool?

Regards,
Lennart

**Comment by edg1983** [comment]
Hi Lennart,

Which kind of information you are interested in exactly?

Some additional information are needed by green-varan to compute variant level, namely prediction scores, TFBS / DNase / UCNE annotations and variant AF. These information need to be present before running green-varan and they are preserved in the output VCF. You can either annotate them by yourself and then set the corresponding fields in the config JSON or you can take advantage of the Nextflow pipeline included in the release to automate the process.

On the other hand, if you need more details on the GREEN-DB annotations reported for a variant (or multiple ones) I suggest you first identify a set of interesting candidates by filtering your VCF and then use green-varan querytab to generate an input table that can be used with greendb_query tool to extract additional information from the DB.

**Comment by ljohansson** [comment]
Hi Eduardo,

Thank you for your answer. Do I understand correctly that the greenvaran smallvars command does not annotated the TFBS/DNase/UCNE annotations and variant AF? In that case I will switch back to the workflow. However, in this workflow I had the initial issue with the 'chr' prefix. I will try and see if I can get it to work. If you haven an example command that would be helpful. 

The information that I am interested in would be the 'all' option that is available in the workflow.

**Comment by edg1983** [comment]
That's correct, greenvaran itself do not add extra annotations outside GREEN-DB, but rely on annotations already in the VCF. These can be added automatically using the workflow.

In your case, until I fix the chromosome issue in the code, you can try something like

```
nextflow workflow/main.nf \
    -profile local \
    --input input_file.vcf.gz \
    --build GRCh37 \
    --out results \
    --scores best \
    --regions best \
    --AF \
    --greenvaran_config config/prioritize_smallvars.json \
    --greenvaran_dbschema config/greendb_schema_v2.5.json \
    --greenvaran_options '--chrom 1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y'
```

You have to ensure to have your updated GREEN-DB bed.gz file (the one without chr prefix) in `resources/GRCh37` and the file must be named `GRCh37_GREEN-DB.bed.gz` with index `GRCh37_GREEN-DB.bed.gz.csi`

The annotations upstream of greenvaran use GRCh37 files and are agnostic to chr prefix so they should work fine.

**Comment by ljohansson** [comment]
Thank you. I will try (currently running). 
Will the command work using 
--scores all \
--regions all\

When using those options the workflow wants to annotate using GRCh38 annotation sources.

**Comment by edg1983** [comment]
There was a typo in sources for TAD files. I've just pushed a fix, see #5  , so it should work now if you clone the latest commit

**Comment by edg1983** [comment]
See my answer to your other issue. There was a typo in the file describing
GRCh37 resources so TAD source always pointed to the GRCh38 file.
I've just pushed a fix so you can clone the repo and it should work.

The others files should have been OK even in the previous version so I
think you see the GRCh38 message only for TAD
